test/test_python_dispatch.py

```diff
@@ -55,6 +55,23 @@ def wrap(e):
         logging.getLogger("LoggingTensor").info(f"{func.__module__}.{func.__name__}", args, rs)
         return rs
 
+class FiniteTensor(torch.Tensor):
+    @staticmethod
+    def __new__(cls, elem, *args, **kwargs):
+        assert torch.isfinite(elem).all()
+        return torch.Tensor._make_subclass(cls, elem, elem.requires_grad)
+
+    def __repr__(self):
+        return f"FiniteTensor({super().__repr__})"
+
+    @classmethod
+    def __torch_dispatch__(cls, func, types, args=(), kwargs=None):
+        # TODO: also explicitly recheck invariants on inplace/out mutation
+        assert not kwargs
```

Comment:
At some point do we want to figure out how to parse the kwargs correctly before calling into __torch_dispatch__? That sounds pretty BC-breaking 😛  (and better to do sooner rather than later?)

Reply:
yeahhhhh